deepmd/pt/model/model/__init__.py

```
@@ -155,19 +158,55 @@ def get_spin_model(model_params: dict) -> SpinModel:
+            require_shared_type_map=False,
+        )
+        model_params["models"] = list(shared_config["model_dict"].values())
+        if "type_map" not in model_params:
+            for idx, sub_model_params in enumerate(model_params["models"]):
+                if "type_map" not in sub_model_params:
+                    raise ValueError(
+                        f"Linear sub-model {idx} must define type_map when "
+                        "linear_ener has no top-level type_map."
+                    )
+            first_type_map = model_params["models"][0]["type_map"]
+            for idx, sub_model_params in enumerate(model_params["models"][1:], start=1):
```

Ruff (0.15.17)

[warning] Loop control variable `idx` not used within loop body

(B007)